generators/import-jdl/index.js

```diff
@@ -154,93 +129,85 @@ module.exports = class extends BaseGenerator {
                     }
                     this.error(`Error while parsing applications and entities from the JDL ${error}`);
                 }
+            }
+        };
+    }
+
+    get writing() {
+        return {
+            generateApplications() {
+                if (!shouldGenerateApplications(this) || this.importState.exportedApplications.length === 0) {
+                    return;
+                }
+                if (this.importState.exportedApplications.length === 1) {
+                    try {
+                        generateApplicationFiles(
+                            this,
+                            this.importState.exportedApplications[0]
+                        );
+                    } catch (error) {
+                        this.error(`While generating applications from the parsed JDL\n${error}`);
+                    }
+                } else {
+                    // sub-folder generation, not yet handled
+                    this.applicationsLeftToGenerate = this.importState.exportedApplications
```

Comment:
what is this usecase?

Reply:
You could say it's temporary. Right now, the standard way of importing works. This is just to warn the user that generating entities/apps from one folder to another is not yet supported (I don't know if it's possible)

Reply:
lets not support that at all, thats why I wanted to remove the path option in new JDL as well, as this could be easily abused and adds a lot of edge cases and possible bugs, lets keep it simple. Also it would be an issue in JHipster online if we support generating apps in other folders, our users are already familiar with the restrictions of yeoman and hence lets just stick to that

Reply:
Oh wait. I may have just understood the comment. I strongly disagree. Generating multiple apps is a must-have. Just so that we are on the same page here
  - there is no app path anymore
  - the chosen way to generate more than one app (instead of having a ton of JDL files) is to create folders so that any subfolder will be at `./<SUBFOLDER_i>` (the generator already handles not having paths as name)
  - a security would be not to generate anything if a baseName is an existing file (one can easily check this with the path module)

If we don't call the subgens in any of the subfolders, that's completely fine by me (that's why I wanted to inform the users of the remaining apps & entities to generate). But not supporting creating these subfolders with `.yo-rc.json` and `.jhipster` files in them would be a mistake in my opinion.

Reply:
@MathieuAA I understand, but for JHipster Online we need to re-think the whole system, and that's just not possible now

Reply:
ok, may be we were talking about different things here. Of course we have to support generating multiple apps form the same JDL, but then the only real use case for that is microservices and I dont see why we need to support that for a monolith, as IMO for monolith it would be a bad practice to have 2 apps on the same JDL.

Good to know that the path option is removed that means we have control of deciding where the path would be so I imagine the below use cases.

1. Entites in a JDL
  - same behavior as of today
2. single monolith app in a JDL with or without entities
  - `mkdir myApp && cd myApp`
  - Execute `jhipster import-jdl app.jh` and we create the below inside myApp folder
  - -- .yo-rc.json
    -- .jhipster
  - We call the app generator by specifying --with-entities flag
  - app is created in the `myApp` folder

3. Multiple microservice in a JDL with entities (say gateway, ms1, ms2)
  - `mkdir myMsApp && cd myMsApp`
  - Execute `jhipster import-jdl app.jh` and we create the below inside myApp folder
  - -- gateway
        -- .yo-rc.json
        -- .jhipster
    -- ms1
        -- .yo-rc.json
        -- .jhipster
    -- ms2
        -- .yo-rc.json
        -- .jhipster
  - We call the app generator by specifying --with-entities flag inside each sub folder, i.e `gateway, ms1 and ms2`
  - app is created in the sub folders accordingly

Now for case 3 the way yeoman works might be a challenge, but we have 2 options for that
1. Since we use our own CLI, we could cd into the folders we want and then execute the generators in the folders. This means import-jdl will not be a sub generator anymore and will be a CLI feature instead (to end user it doesnt matter as they will use it in the sameway, but a sideeffect would be that people wont be able to do `yo jhipster:import-jdl` anymore
2. Do some hack to cd into the folders before composing the app generators and cd out once done

Reply:
The first option looks good, but it implies having to work a bit to achieve that.
Let's not bother ourselves about how we generate these multiple apps. Anyone can temporarily `cd` inside the folders and do a `yo jhipster`. I'd rather have the app composition working.
I can, in the meantime, issue a warning that a JDL file contains multiple apps (to inform it's a bad practise).